### PR TITLE
chore: Small optimization of the actor graph interpreter

### PR DESCRIPTION
**Pull request by johanandren**
Just duplicating what we do for typed actors. Didn't bench to see if it makes any difference yet.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala [hunk 1]
```diff
@@ -812,23 +816,31 @@ import akka.util.OptionVal
...
+        sender() ! StreamSnapshotImpl(
+          self.path,
+          activeInterpreters.map(shell => shell.toSnapshot.asInstanceOf[RunningInterpreter]).toSeq,
+          newShells.map(shell => shell.toSnapshot.asInstanceOf[UninitializedInterpreter]))
+
+      case other => unhandled(other)
+    }
   }
 
+  override def receive: Receive = ActorGraphInterpreter.DummyReceive
```
Comment (He-Pin) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala:842]: Is there any reason not inline the value definition?

Comment (johanandren) [reply akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala:842]: Actually only called once, during actor startup, so not important to optimize

**Comment by johanandren** [comment]
Benchmarked and no mentionable improvement on neither x86 nor aarch64. So let's not.

One surprise though is that the stream async boundaries are extremely more costly on arm64 for some reason, modded FlowMapBenchmark adding a `.async` around each map, I wonder what that is about:

Mac M1 Max
```
[info] FlowMapBenchmark.flow_map_100k_elements                 1  thrpt    5  3091022,317 ± 296855,355  ops/s
[info] FlowMapBenchmark.flow_map_100k_elements                 5  thrpt    5   769869,867 ±  19422,731  ops/s
[info] FlowMapBenchmark.flow_map_100k_elements                10  thrpt    5   848269,776 ±  26097,537  ops/s
```

Intel i5:
```
[info] FlowMapBenchmark.flow_map_100k_elements                 1  thrpt    5  3146755.247 ± 144071.725  ops/s
[info] FlowMapBenchmark.flow_map_100k_elements                 5  thrpt    5  1832234.561 ±  24307.655  ops/s
[info] FlowMapBenchmark.flow_map_100k_elements                10  thrpt    5  1804830.299 ±  21332.813  ops/s
```

**Comment by He-Pin** [comment]
What the bytecode says？

**Comment by johanandren** [comment]
I'll see if I have time to investigate, for now I'll just see it as a difference enough between the platforms that it is not worth benchmarking on arm64 (until we expect there actually are any users not on amd64)

**Pull request closed without merging**